edge-agent/test/Microsoft.Azure.Devices.Edge.Agent.DiagnosticsComponent.Test/MetricsFileStorageTest.cs

```diff
@@ -0,0 +1,137 @@
+// Copyright (c) Microsoft. All rights reserved.
+
+namespace Microsoft.Azure.Devices.Edge.Agent.DiagnosticsComponent.Test
+{
+    using System;
+    using System.Collections.Generic;
+    using System.IO;
+    using System.Linq;
+    using Microsoft.Azure.Devices.Edge.Util;
+    using Microsoft.Azure.Devices.Edge.Util.Test.Common;
+    using Moq;
+    using Xunit;
+
+    public class MetricsFileStorageTest : IDisposable
+    {
+        DateTime fakeTime;
+        ISystemTime systemTime;
+        TempDirectory tempDirectory = new TempDirectory();
```

Comment:
Constructor and dispose method will be called for each test; therefore tempDirectory always contain 1 entry when running these tests.  I would suggest to create a static folder name "metricsFileStorageTest" under temp folder; and clean it up in dispose method; don't need TempDirectory at all.

Reply:
I like it because it abstracts the work of cleaning up the directory and can be used in multiple tests.